src/platform/plugins/shared/embeddable/common/embeddable_content_management/registry.ts

```diff
@@ -40,16 +40,19 @@ class EmbeddableContentManagementRegistry<IsAsync extends true | false = false>
     this.registry.set(id, getDefinition);
   };
 
-  public getContentManagementDefinition(id: string): MaybeDefinitionSyncOrAsync<IsAsync> {
+  public getContentManagementDefinition: (id: string) => MaybeDefinitionSyncOrAsync<IsAsync> = (
+    id
+  ) => {
     const getDefinition = this.registry.get(id);
     if (!getDefinition) {
       return undefined as IsAsync extends true ? Promise<undefined> : undefined;
     }
     return getDefinition();
-  }
+  };
 }
 
 export class EmbeddableContentManagementRegistryPublic extends EmbeddableContentManagementRegistry<true> {
+  // @ts-expect-error Overload arrow function with member function to correctly bind `this` to superclass
```

Comment:
Note that doing something like `public baseGetContentManagementDefinition` in the superclass to avoid this Typescript error will break `this` binding all over again. We really do need `getContentManagementDefinition` to be an arrow function in the superclass and then rebind it with an overload in the inheriting classes, and Typescript does NOT like this at all, but sometimes Typescript can be wrong.